lib/Serialization/ModuleFile.cpp

```diff
@@ -339,8 +346,18 @@ class ModuleFile::DeclTableInfo {
   }
 
   static internal_key_type ReadKey(const uint8_t *data, unsigned length) {
-    // TODO: Handle special names
-    return StringRef(reinterpret_cast<const char *>(data), length);
+    uint8_t kind = endian::readNext<uint8_t, little, unaligned>(data);
+    switch (kind) {
+    case (uint8_t)DeclBaseName::Kind::Normal: {
```

Comment:
The ClangImporter one is fine, because that format's never intended to be persistent.